server/remote_cache/byte_stream_server/byte_stream_server.go

```diff
@@ -445,20 +445,28 @@ type zstdDecompressor struct {
 // Note that writes are not matched one-to-one, since the compression scheme may
 // require more than one chunk of compressed data in order to write a single
 // chunk of decompressed data.
-func NewZstdDecompressor(writer io.Writer) io.WriteCloser {
```

Comment:
could this struct be moved to util/compression?

Reply:
sg, done. Moved the compressor there too, and passing in the intermediate buffers as params since they are bytestream-specific.